services/ui-src/src/components/fields/DynamicField.tsx

```diff
@@ -80,7 +104,8 @@ export const DynamicField = ({ name, label, ...props }: Props) => {
             />
             <Box sx={sx.removeBox}>
               <button
-                onClick={() => removeRecord(index)}
+                type="button"
+                onClick={() => openDeleteProgramModal(index)}
```

Comment:
Question: Could we pass the entire field object through here and just set it as the selected one directly?

Reply:
We could, though we really just need the index in this case. If we needed the name of the program then definitely we should pass it through